### ISSUE TITLE
Error when trying to require nodegit

### ISSUE THREAD

**Issue by NoHomey** [nodegit/nodegit#791]
I'm using node 5.1.0 and npm 3.3.12 and I'm getting the following error when i try to require nodegit
This is how i require it in my Cakefile :
```
{Clone} = require "nodegit"
```
```
module.js:340
    throw err;
    ^

Error: Cannot find module '../build/Debug/nodegit.node'
  at Function.Module._resolveFilename (module.js:338:15)
  at Function.Module._load (module.js:289:25)
  at Module.require (module.js:366:17)
  at require (module.js:385:17)
  at Object.<anonymous> (/home/ivo/ivo/diplomna3/node_modules/nodegit/lib/nodegit.js:16:12)
  at Module._compile (module.js:425:26)
  at Object.Module._extensions..js (module.js:432:10)
  at Module.load (/usr/lib/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
  at Function.Module._load (module.js:313:12)
  at Module.require (module.js:366:17)
  at require (module.js:385:17)
  at Object.action (/home/ivo/ivo/diplomna3/Cakefile:85:18)
  at helpers.extend.invoke (/usr/lib/node_modules/coffee-script/lib/coffee-script/cake.js:44:26)
  at Object.exports.run (/usr/lib/node_modules/coffee-script/lib/coffee-script/cake.js:70:20)
  at Object.<anonymous> (/usr/lib/node_modules/coffee-script/bin/cake:7:38)
  at Module._compile (module.js:425:26)
  at Object.Module._extensions..js (module.js:432:10)
  at Module.load (module.js:356:32)
  at Function.Module._load (module.js:313:12)
  at Function.Module.runMain (module.js:457:10)
  at startup (node.js:138:18)
  at node.js:974:3
```

**Comment by saper** [comment]
How was nodegit installed? Does `npm install nodegit` work in a clean directory?

**Comment by NoHomey** [comment]
Install.....

```│ ├─┬ form-data@1.0.0-rc3 
│ │ └── async@1.5.0 
│ ├─┬ har-validator@2.0.2 
│ │ ├─┬ chalk@1.1.1 
│ │ │ ├── ansi-styles@2.1.0 
│ │ │ ├─┬ has-ansi@2.0.0 
│ │ │ │ └── ansi-regex@2.0.0 
│ │ │ ├── strip-ansi@3.0.0 
│ │ │ └── supports-color@2.0.0 
│ │ ├─┬ commander@2.9.0 
│ │ │ └── graceful-readlink@1.0.1 
│ │ ├─┬ is-my-json-valid@2.12.3 
│ │ │ ├── generate-function@2.0.0 
│ │ │ ├─┬ generate-object-property@1.2.0 
│ │ │ │ └── is-property@1.0.2 
│ │ │ └── jsonpointer@2.0.0 
│ │ └─┬ pinkie-promise@1.0.0 
│ │   └── pinkie@1.0.0 
│ ├─┬ hawk@3.1.2 
│ │ ├── boom@2.10.1 
│ │ ├── cryptiles@2.0.5 
│ │ ├── hoek@2.16.3 
│ │ └── sntp@1.0.9 
│ ├─┬ http-signature@1.1.0 
│ │ ├── assert-plus@0.1.5 
│ │ ├─┬ jsprim@1.2.2 
│ │ │ ├── extsprintf@1.0.2 
│ │ │ ├── json-schema@0.2.2 
│ │ │ └── verror@1.3.6 
│ │ └─┬ sshpk@1.7.0 
│ │   ├── asn1@0.2.3 
│ │   ├── assert-plus@0.2.0 
│ │   ├── dashdash@1.10.1 
│ │   ├─┬ ecc-jsbn@0.0.1 
│ │   │ └── jsbn@0.0.0  (git+https://github.com/rynomad/jsbn.git#bb522b0124f75424f89d49446c40a87111942c7b)
│ │   ├── jodid25519@1.0.2 
│ │   ├── jsbn@0.1.0 
│ │   └── tweetnacl@0.13.2 
│ ├── is-typedarray@1.0.0 
│ ├── isstream@0.1.2 
│ ├── json-stringify-safe@5.0.1 
│ ├─┬ mime-types@2.1.7 
│ │ └── mime-db@1.19.0 
│ ├── node-uuid@1.4.7 
│ ├── oauth-sign@0.8.0 
│ ├── qs@5.2.0 
│ ├── stringstream@0.0.5 
│ ├── tough-cookie@2.2.1 
│ └── tunnel-agent@0.4.1 
├─┬ tar@2.2.1 
│ ├── block-stream@0.0.8 
│ └── inherits@2.0.1 
└─┬ which-native-nodish@1.1.3 
  ├── find-parent-dir@0.3.0 
  └─┬ nodegit-promise@1.0.2 
    └── asap@1.0.0 


[nodegit] Configuring libssh2.
{ [Error: Command failed: /home/ivo/ivo/node_modules/nodegit/vendor/libssh2/configure
configure: error: cannot find OpenSSL or Libgcrypt,
try --with-libssl-prefix=PATH or --with-libgcrypt-prefix=PATH
]
  killed: false,
  code: 1,
  signal: null,
  cmd: '/home/ivo/ivo/node_modules/nodegit/vendor/libssh2/configure' }
configure: error: cannot find OpenSSL or Libgcrypt,
try --with-libssl-prefix=PATH or --with-libgcrypt-prefix=PATH

/home/ivo/ivo
└── nodegit@0.5.0 

npm WARN ENOENT ENOENT: no such file or directory, open '/home/ivo/ivo/package.json'
npm WARN EPACKAGEJSON ivo No description
npm WARN EPACKAGEJSON ivo No repository field.
npm WARN EPACKAGEJSON ivo No README data
npm WARN EPACKAGEJSON ivo No license field.
ivo@ivo-dell:~/ivo/test$ node
> git = require("nodegit")
Error: Cannot find module '../build/Debug/nodegit.node'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:289:25)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/home/ivo/ivo/node_modules/nodegit/lib/nodegit.js:16:12)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
```


I have installed libgit2, libssl & libgcrypt and all are in my PATH using ubuntu 14.04

**Comment by saper** [comment]
It also seems that your `node` has been configured in the `Debug` mode (check with `node -p process.config.target_defaults`).

**Comment by NoHomey** [comment]
{ cflags: [],
  default_configuration: 'Release',
  defines: [],
  include_dirs: [],
  libraries: [] }

**Comment by saper** [comment]
Anyway, your module didn't install properly. Do you know why `node-pre-gyp` could not fetch the binary for you?

**Comment by NoHomey** [comment]
Well I'm coding on Ubuntu with Dell overlays with amd64 arch.
This is the only think that comes up to me .
Will test it on my desktop wich is still with node 0.12 and inform you what happens there

**Comment by NoHomey** [comment]
Same

**Comment by saper** [comment]
Another question, do you have openssl-dev installed?

**Comment by NoHomey** [comment]
Nope i didn't but installing it solved the problem. I'm thinking that if there are  dependencies would be nice. I have seen it in through other big node packages  sites :)

**Comment by saper** [comment]
openssl and libgit2 are not JavaScript dependencies, you can't install them via npm.

Can you post your full log of `npm install nodegit` on gist.github.com? and also your `npm-debug.log` file

**Comment by NoHomey** [comment]
Installing libssl-dev solved the problem so I'm closing this issue

**Comment by n8behavior** [comment]
yup, `sudo apt install libssl-dev` fixes for Ubuntu 16.04
